### PR TITLE
Pipe: add compression level config for connector ZSTD compressor

### PR DESCRIPTION
**Pull request by DanielWang2035**
Currently, ZSTD compressor in the TsFile module does not support the configuration of compression level. We hope to support this in the Pipe RPC compression.

### CODE REVIEW

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/config/PipeConfig.java [hunk 1]
```diff
@@ -143,6 +143,10 @@ public double getPipeRemainingTimeCommitRateSmoothingFactor() {
     return COMMON_CONFIG.getPipeRemainingTimeCommitRateSmoothingFactor();
   }
 
+  public int getPipeConnectorRPCCompressionZSTDCompressorLevel() {
+    return COMMON_CONFIG.getPipeConnectorRPCCompressionZSTDCompressorLevel();
```
Comment (Caideyipi) [inline iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/config/PipeConfig.java:147]: Print it.

**Review by Caideyipi — APPROVED** [review]

**Review by SteveYurongSu — COMMENTED** [review]
Please add some ITs :)

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/config/constant/PipeConnectorConstant.java [hunk 2]
```diff
@@ -177,6 +179,14 @@ public class PipeConnectorConstant {
                   CONNECTOR_COMPRESSOR_ZSTD,
                   CONNECTOR_COMPRESSOR_LZMA2)));
 
+  public static final String CONNECTOR_COMPRESSOR_ZSTD_LEVEL_KEY =
+      "connector.compressor.zstd.level";
+  public static final String SINK_COMPRESSOR_ZSTD_LEVEL_KEY = "sink.compressor.zstd.level";
+  public static final int CONNECTOR_COMPRESSOR_ZSTD_LEVEL_DEFAULT_VALUE =
+      Zstd.defaultCompressionLevel();
+  public static final int CONNECTOR_COMPRESSOR_ZSTD_LEVEL_MIN_VALUE = Zstd.minCompressionLevel();
```
Comment (DanielWang2035) [inline iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/config/constant/PipeConnectorConstant.java:187]: Zstd.minCompressionLevel() will return -131072 here, I don't know how it works.

Comment (SteveYurongSu) [reply iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/config/constant/PipeConnectorConstant.java:187]: It seems that the return value of Zstd.minCompressionLevel() is as expected..

See: https://github.com/luben/zstd-jni/issues/132

Comment (DanielWang2035) [reply iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/config/constant/PipeConnectorConstant.java:187]: ok.

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/connector/protocol/IoTDBConnector.java [hunk 3]
```diff
@@ -151,6 +157,21 @@ public void validate(PipeParameterValidator validator) throws Exception {
             CONNECTOR_LOAD_BALANCE_STRATEGY_SET, loadBalanceStrategy),
         loadBalanceStrategy);
 
+    final int zstdCompressionLevel =
+        parameters.getIntOrDefault(
+            Arrays.asList(CONNECTOR_COMPRESSOR_ZSTD_LEVEL_KEY, SINK_COMPRESSOR_ZSTD_LEVEL_KEY),
+            CONNECTOR_COMPRESSOR_ZSTD_LEVEL_DEFAULT_VALUE);
+    validator.validate(
+        arg ->
+            zstdCompressionLevel >= CONNECTOR_COMPRESSOR_ZSTD_LEVEL_MIN_VALUE
```
Comment (Caideyipi) [inline iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/connector/protocol/IoTDBConnector.java:166]: This logic is runnable but the “arg” is not used...

**Review by SteveYurongSu — APPROVED** [review]

**Review by SteveYurongSu — APPROVED** [review]

**Pull request merged**